ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java

```diff
@@ -2147,7 +2151,8 @@ protected synchronized final int prepare (boolean reportHeuristics)
 
             ActionManager.manager().remove(get_uid());
 
-            return TwoPhaseOutcome.PREPARE_ONE_PHASE_COMMITTED;
+            return actionStatus == ActionStatus.ABORTED
```

Comment:
On the current diff now :)

One option is to change onePhasePrepare API to return the right status https://github.com/jbosstm/narayana/blob/master/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/BasicAction.java#L2322. That would be a public API change so not advisible for this release but might be something to consider for future.

How about returning TwoPhaseOutcome.PREPARE_NOTOK if the 1PC fails so keep the code on line 1501 as is?